From: "Yan, Zheng" <zyan@redhat.com>
Date: Fri, 29 Apr 2016 15:58:32 +0800
Subject: ceph: remove unnecessary checks in __dcache_readdir
Git-commit: 1cd42a429174689c0df1c37b642654a1ab4d1506
Patch-mainline: v4.7-rc1
References: FATE#322288

we never add snapdir and the hidden .ceph dir into readdir cache

Signed-off-by: Yan, Zheng <zyan@redhat.com>
Acked-by: Luis Henriques <lhenriques@suse.com>
---
 fs/ceph/dir.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/fs/ceph/dir.c b/fs/ceph/dir.c
index cdea450c5902..734508be1526 100644
--- a/fs/ceph/dir.c
+++ b/fs/ceph/dir.c
@@ -228,8 +228,6 @@ static int __dcache_readdir(struct file *file,  struct dir_context *ctx,
 		spin_lock(&dentry->d_lock);
 		if (di->lease_shared_gen == shared_gen &&
 		    d_really_is_positive(dentry) &&
-		    ceph_snap(d_inode(dentry)) != CEPH_SNAPDIR &&
-		    ceph_ino(d_inode(dentry)) != CEPH_INO_CEPH &&
 		    fpos_cmp(ctx->pos, di->offset) <= 0) {
 			emit_dentry = true;
 		}

